test-infra/kf-v0-4-n00/gcp_config/cluster-kubeflow.yaml

```diff
@@ -0,0 +1,97 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
```

Comment:
Why duplicate config files in test-infra/kf-v0-4-n00/gcp_config?
Though I believe it will be most likely stable and backward compatible.

Reply:
kubeflow/testing#95 is tracking automating the invocation of this script.

test-infra-kf-v0-4-n00 is the kfctl app created by running kfctl. So we are just checking the configs specific to a particular kubeflow deployment.

create_kf_instance.py is just a very thin
https://github.com/kubeflow/testing/pull/269/files#diff-bfc99eeada8dcde79ce1e19a295cfb0b
wrapper around kfctl.sh

Really the only thing its doing is adding logic to
1. pick the name for the cluster 
1. Load OAuthClient ID/Secret form a GCP bucket

I agree with you we should consider refactoring the code to potentially share code. Specifically, 
we might want to reuse the logic in test_deploy_app.py to reuse SSL certificates.

I added a comment here: about resuing the logic for SSL certs.
https://github.com/kubeflow/testing/issues/95#issuecomment-450440423

I'm not sure if it makes sense to use the web app as opposed to kfctl to deploy the clusters. I think one advantage of using kfctl is that all the config files are local. With click to deploy we have to do a manual step to get access to the configs and configure them.